realm/realm-library/src/androidTest/java/io/realm/RealmResultsTests.java

```diff
@@ -1311,6 +1311,21 @@ public void distinct() {
     }
 
     @Test
+    public void distinct_resultsWithTableView() {
+        final long numberOfBlocks = 25;
+        final long numberOfObjects = 10; // must be greater than 1
+        populateForDistinct(realm, numberOfBlocks, numberOfObjects, false);
+
+        RealmResults<AnnotationIndexTypes> rootResults = realm.where(AnnotationIndexTypes.class).findAll();
+        assertEquals("Total Count", numberOfBlocks * numberOfBlocks * numberOfObjects, rootResults.size());
+        RealmResults<AnnotationIndexTypes> parentResults = rootResults.distinct("indexDate");
```

Comment:
I would add some more comments in this code, preferably above each `RealmResults` to indicate what is happening. Right now the code is very dense and hard to read at a glance.